src/utils/check-pr/check-pr.py

```diff
@@ -210,10 +212,61 @@ def deploy(self, filename: str) -> None:
             print(msg)
             subprocess.check_call(cmd, shell=True)
 
+    def register(self) -> None:
+        sp_name = "sp_" + self.instance
+        print(f"registering {sp_name} to {self.instance}")
+
+        venv = "register-venv"
+        subprocess.check_call(f"python3 -mvenv {venv}", shell=True)
+        pip = venv_path(venv, "pip")
+        py = venv_path(venv, "python3")
+
+        az_cmd = ["az", "account", "show", "--query", "id", "-o", "tsv"]
+        subscription_id = subprocess.check_output(az_cmd, encoding="UTF-8")
+        subscription_id = subscription_id.strip()
+
+        commands = [
+            ("installing prereqs", f"{pip} install -q -r requirements.txt"),
+            (
+                "running cli registration",
+                (
+                    f"{py} ./deploylib/registration.py create_cli_registration "
+                    f"{self.instance} {subscription_id}"
+                    f" --registration_name {sp_name}"
+                ),
+            ),
+            (
```

Comment:
why do we need to assign the cli role here 
`create_cli_registration` should have already assigned the correct role to the application